### PR TITLE
Add stack.yaml files for GHC 7.8.4 and 7.10.3. Closes #2693.

### LINKED ISSUE #2693: Missing stack.yaml files for GHC 7.8.4 and 7.10.3
After installing [2.5.3 RC 1](http://www1.eafit.edu.co/asr/tmp-asr/Agda-2.5.2.20170813.tar.gz) I realised that the Stackage `*.yaml` files for GHC 7.8.4 and 7.10.3 are missing. Are we going to include these files for each version supported of GHC? If so, we also should add the corresponding Travis tests.

### PR DESCRIPTION
**Pull request by favonia**
Still waiting for the travis jobs, and the one for 7.10.3 seems to work. Checking the one for 7.8.4 now.

This will fix #2693.

### CODE REVIEW

**Comment by favonia** [comment]
@asr According to the travis jobs for my local copies, I believe everything is working now.

**Comment by favonia** [comment]
PS: One Travis job of the last commit failed probably because of some network connection problem.

**Comment by asr** [comment]
I just restarted the failing job.

File: .travis.yml [hunk 1]
```diff
@@ -193,12 +209,19 @@ install:
...
+# outside the designated snapshot. [Issue commercialhaskell/stack #595]
+
 # N.B. that we use the `--force-reinstalls` option [Issue 1520].
 
   - if [[ $TEST = "MAIN" || $TEST = "HADDOCK" ]]; then
        make CABAL_OPTS='--only-dependencies --force-reinstalls' install-bin;
     elif [[ $TEST = "STACKAGE" ]]; then
-         stack build $ARGS --no-terminal --test --only-dependencies;
+       if [[ $MANUAL_INSTALL_BUILD_TOOLS = "YES" ]]; then
+           stack install $ARGS --no-terminal alex cpphs happy;
```
Comment (asr) [inline .travis.yml:225]: How know `stack` which versions of Alex, cpphs and Happy should install?

Comment (favonia) [reply .travis.yml:225]: The version of `cpphs` is specified in the `yaml` files. I then realized we need to install `alex` and `happy` too, but I do not fully understand why stack cannot figure out the correct versions to install for `lts-2.22` and `lts-6.35`.

Comment (favonia) [reply .travis.yml:225]: The issue https://github.com/commercialhaskell/stack/issues/1427 seems related, and then it points back to https://github.com/commercialhaskell/stack/issues/595. Thus I leave the comments like that. I do not know what is wrong with the resolver for build tools. In particular, I did not specify the versions of happy and alex in the `yaml` file.

**Comment by favonia** [comment]
I updated the comments in `.travis.yaml` a little bit.

File: .travis.yml (hunk 1 above)
Comment (favonia) [reply .travis.yml:225]: FYI: the version matrix is
```
lts   happy   alex   cpphs
2.22  1.19.5  3.1.4  1.19.2
6.35  1.19.5  3.1.7  1.20.5
last  1.19.5  3.2.1  1.20.8
```
and (theoretically) we only have to overwrite the version of `cpphs`.

**Comment by asr** [comment]
@favonia, thank you!

**Comment by gallais** [comment]
We could maybe use [`travis_retry`](https://docs.travis-ci.com/user/common-build-problems/#travis_retry) for the parts of the installation process that are subject to network connectivity issues?

**Comment by asr** [comment]
Good idea. Could you write a PR, please?

**Comment by gallais** [comment]
Sure thing.

**Pull request merged**